server/authz/policy.rego

```diff
@@ -211,47 +216,76 @@ allow {
 # Hosts
 ##
 
+# allowed_read_roles evaulates which roles are allowed for read based on the given action.
+allowed_read_roles(action, base_roles, extra_roles) = result {
+	action == selective_read
+	result := base_roles | extra_roles
+} else = result {
+	action == read
+	result := base_roles
+} else = result {
+	result := null
+}
```

Comment:
:tada: nice job figuring out the weird syntax of rego to do this :D

Reply:
thank you!! definitely took some trial and error 😅 

I'm still not sold on this being the best way to approach this problem to be honest, if anyone has other ideas I can give them a try.

Reply:
I think it's more maintainable going forward, as more special cases will likely be added. My only worry I guess is a semantic one, if it's ok to have `gitops` have "read" accesses when we say [it's a "write-only" role](https://fleetdm.com/docs/using-fleet/manage-access#gitops), but really in the end it really is only a semantic concern as it has to be able to make those reads, however we implement it.